crates/ruff_linter/src/rules/pylint/helpers.rs

```diff
@@ -307,3 +307,30 @@ pub(super) fn is_known_dunder_method(method: &str) -> bool {
             | "_generate_next_value_"
     )
 }
+
+/// Returns `true` if the method is a known dunder method that is only allowed in Python 2.
```

Comment:
Calling `is_known_dunder_method` first and then `is_deprecated_dunder_method_in_python3` has the downside that it results in two string matches. We can avoid this by:

* Introduce a new `DunderMethodKind` with two variants: `Known`, `Removed`
* Add a `dunder_method_kind(name: &str) -> Option<DunderMethodKind>` 
* Change `is_known_dunder_method` to `dunder_method_kind(name) == DunderMethodKind::Known`